yt_dlp/networking/impersonate.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass, fields
+from typing import Any
+
+from .common import RequestHandler, register_preference
+from .exceptions import UnsupportedRequest
+from ..compat.types import NoneType
+from ..utils import classproperty
+from ..utils.networking import std_headers
+
+
+@dataclass(order=True, frozen=True)
+class ImpersonateTarget:
```

Comment:
Why is `client=None, version=xxx` etc allowed?

Reply:
keep impl simple and generic. it's something undocumented and undefined, so doesn't really matter imo